torchao/prototype/float8nocompile/test/test.py

```diff
@@ -0,0 +1,73 @@
+from copy import deepcopy
+
+import pytest
+import torch
+import torch.nn as nn
+
+from torchao.float8.float8_linear_utils import convert_to_float8_training
+
+from torchao.prototype.float8nocompile.float8nocompile_linear_utils import (
+    convert_to_float8_nocompile_training,
+)
+
+from torchao.utils import TORCH_VERSION_AT_LEAST_2_5
+
+if not TORCH_VERSION_AT_LEAST_2_5:
+    raise AssertionError("torchao.float8 requires PyTorch version 2.5 or greater")
+
+
+class TestModel(nn.Module):
+    def __init__(self):
+        super().__init__()
+        self.layers = nn.Sequential(
+            nn.Linear(32, 64, bias=False),
+            nn.Linear(64, 32, bias=False),
+        )
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        return self.layers(x)
+
+
+@pytest.fixture
+def model1():
+    torch.manual_seed(0)
+    return TestModel()
+
+
+@pytest.fixture
+def model2():
+    torch.manual_seed(0)
+    return TestModel()
+
+
+def test_model_weights_and_gradients(model1, model2):
+    assert torch.cuda.is_available()
+    device = torch.device("cuda")
+
+    model1 = model1.to(torch.bfloat16).to(device)
+    model2 = model2.to(torch.bfloat16).to(device)
+
+    # compare production float8 linear conversion with no-compile version
+    convert_to_float8_nocompile_training(model1)
+    convert_to_float8_training(model2)
```

Comment:
readability nit: order the baseline first and the experiment second, IMO that's easier to read for the average code reader.  Right now it's the other way around.